File: model-client/src/main/java/org/modelix/model/api/ITreeChangeVisitor.java
package org.modelix.model.api;

public interface ITreeChangeVisitor {
  void containmentChanged(long nodeId);
  void childrenChanged(long nodeId, String role);
  void referenceChanged(long nodeId, String role);
  void propertyChanged(long nodeId, String role);
  void userObjectChanged(long nodeId, Object key);
  void nodeRemoved(long nodeId);
  void nodeAdded(long nodeId);
  void nodeLoaded(long nodeId);
  void nodeUnloaded(long nodeId);
}

File: model-client/src/main/java/org/modelix/model/api/ITreeChangeVisitor.kt
package org.modelix.model.api

interface ITreeChangeVisitor {
    fun containmentChanged(nodeId: Long)
    fun childrenChanged(nodeId: Long, role: String?)
    fun referenceChanged(nodeId: Long, role: String?)
    fun propertyChanged(nodeId: Long, role: String?)
    fun userObjectChanged(nodeId: Long, key: Any?)
    fun nodeRemoved(nodeId: Long)
    fun nodeAdded(nodeId: Long)
    fun nodeLoaded(nodeId: Long)
    fun nodeUnloaded(nodeId: Long)
}